ClangTidy warnings for `misc-use-internal-linkage`

```
/Users/jviotti/Projects/core/src/core/json/json.cc:18:6: error: variable 'parse_json' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   18 | auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:24:6: error: function 'parse_json' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   24 | auto parse_json(const std::basic_string<JSON::Char, JSON::CharTraits> &input,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:30:6: error: variable 'parse_json' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   30 | auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:37:6: error: function 'parse_json' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   37 | auto parse_json(const std::basic_string<JSON::Char, JSON::CharTraits> &input,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:44:6: error: function 'read_file' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   44 | auto read_file(const std::filesystem::path &path)
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:64:6: error: function 'read_json' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   64 | auto read_json(const std::filesystem::path &path,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:78:6: error: function 'stringify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   78 | auto stringify(const JSON &document,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:84:6: error: function 'prettify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   84 | auto prettify(const JSON &document,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:90:6: error: function 'stringify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   90 | auto stringify(const JSON &document,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:96:6: error: function 'prettify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   96 | auto prettify(const JSON &document,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:102:6: error: function 'operator<<' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  102 | auto operator<<(std::basic_ostream<JSON::Char, JSON::CharTraits> &stream,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/json/json.cc:113:6: error: function 'operator<<' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  113 | auto operator<<(std::basic_ostream<JSON::Char, JSON::CharTraits> &stream,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:140:6: error: function 'get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  140 | auto get(const JSON &document, const Pointer &pointer) -> const JSON & {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:148:6: error: function 'get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  148 | auto get(const JSON &document, const WeakPointer &pointer) -> const JSON & {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:157:6: error: function 'get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  157 | auto get(JSON &document, const Pointer &pointer) -> JSON & {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:165:6: error: function 'try_get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  165 | auto try_get(const JSON &document, const Pointer &pointer) -> const JSON * {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:169:6: error: function 'try_get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  169 | auto try_get(const JSON &document, const WeakPointer &pointer) -> const JSON * {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:173:6: error: function 'get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  173 | auto get(const JSON &document, const Pointer::Token &token) -> const JSON & {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:181:6: error: function 'get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  181 | auto get(const JSON &document, const WeakPointer::Token &token)
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:190:6: error: function 'get' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  190 | auto get(JSON &document, const Pointer::Token &token) -> JSON & {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:198:6: error: function 'set' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  198 | auto set(JSON &document, const Pointer &pointer, const JSON &value) -> void {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:225:6: error: function 'set' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  225 | auto set(JSON &document, const Pointer &pointer, JSON &&value) -> void {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:248:6: error: function 'to_pointer' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  248 | auto to_pointer(const JSON &document) -> Pointer {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:254:6: error: function 'to_pointer' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  254 | auto to_pointer(const std::basic_string<JSON::Char, JSON::CharTraits,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:266:6: error: function 'to_pointer' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  266 | auto to_pointer(const WeakPointer &pointer) -> Pointer {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:279:6: error: function 'to_weak_pointer' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  279 | auto to_weak_pointer(const Pointer &pointer) -> WeakPointer {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:292:6: error: function 'stringify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  292 | auto stringify(const Pointer &pointer,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:299:6: error: function 'stringify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  299 | auto stringify(const WeakPointer &pointer,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonpointer/jsonpointer.cc:306:6: error: function 'stringify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  306 | auto stringify(const PointerTemplate &pointer,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/bundle.cc:144:6: error: function 'bundle' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  144 | auto bundle(sourcemeta::core::JSON &schema, const SchemaWalker &walker,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/bundle.cc:187:6: error: function 'bundle' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  187 | auto bundle(const sourcemeta::core::JSON &schema, const SchemaWalker &walker,
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/frame.cc:414:6: error: function 'operator<<' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  414 | auto operator<<(std::ostream &stream, const SchemaFrame &frame)
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:58:24: error: function 'identify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   58 | auto sourcemeta::core::identify(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:112:24: error: function 'identify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  112 | auto sourcemeta::core::identify(const JSON &schema,
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:153:24: error: function 'anonymize' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  153 | auto sourcemeta::core::anonymize(JSON &schema, const std::string &base_dialect)
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:160:24: error: function 'reidentify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  160 | auto sourcemeta::core::reidentify(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:174:24: error: function 'reidentify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  174 | auto sourcemeta::core::reidentify(JSON &schema,
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:220:24: error: function 'dialect' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  220 | auto sourcemeta::core::dialect(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:234:24: error: function 'metaschema' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  234 | auto sourcemeta::core::metaschema(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:254:24: error: function 'base_dialect' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  254 | auto sourcemeta::core::base_dialect(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:334:24: error: function 'vocabularies' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  334 | auto sourcemeta::core::vocabularies(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:360:24: error: function 'vocabularies' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  360 | auto sourcemeta::core::vocabularies(const SchemaResolver &resolver,
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:453:24: error: function 'schema_format_compare' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  453 | auto sourcemeta::core::schema_format_compare(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:563:24: error: function 'reference_visit' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  563 | auto sourcemeta::core::reference_visit(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:610:24: error: function 'reference_visitor_relativize' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  610 | auto sourcemeta::core::reference_visitor_relativize(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:629:24: error: function 'schema_keyword_priority' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  629 | auto sourcemeta::core::schema_keyword_priority(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:644:24: error: function 'unidentify' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  644 | auto sourcemeta::core::unidentify(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:695:24: error: function 'wrap' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  695 | auto sourcemeta::core::wrap(const sourcemeta::core::JSON::String &identifier)
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/jsonschema.cc:706:24: error: function 'wrap' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  706 | auto sourcemeta::core::wrap(const sourcemeta::core::JSON &schema,
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/official_resolver.in.cc:3:24: error: function 'schema_official_resolver' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
    3 | auto sourcemeta::core::schema_official_resolver(std::string_view identifier)
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/jsonschema/official_walker.cc:3:24: error: function 'schema_official_walker' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
    3 | auto sourcemeta::core::schema_official_walker(
      |                        ^
      | static

/Users/jviotti/Projects/core/src/core/uri/uri.cc:576:11: error: variable 'from_fragment' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  576 | auto URI::from_fragment(std::string_view fragment) -> URI {
      |           ^
      | static

/Users/jviotti/Projects/core/src/core/yaml/yaml.cc:110:6: error: variable 'parse_yaml' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  110 | auto parse_yaml(std::basic_istream<JSON::Char, JSON::CharTraits> &stream)
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/yaml/yaml.cc:117:6: error: function 'parse_yaml' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  117 | auto parse_yaml(const JSON::String &input) -> JSON {
      |      ^
      | static

/Users/jviotti/Projects/core/src/core/yaml/yaml.cc:138:6: error: function 'read_yaml' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
  138 | auto read_yaml(const std::filesystem::path &path) -> JSON {
      |      ^
      | static

/Users/jviotti/Projects/core/src/extension/alterschema/alterschema.cc:10:6: error: function 'contains_any' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   10 | auto contains_any(const Vocabularies &container,
      |      ^
      | static

/Users/jviotti/Projects/core/src/extension/alterschema/alterschema.cc:19:28: error: function 'every_item_is_null' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   19 | template <typename T> auto every_item_is_null(const T &container) -> bool {
      |                            ^
      |                       static

/Users/jviotti/Projects/core/src/extension/alterschema/alterschema.cc:24:28: error: function 'every_item_is_boolean' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage,-warnings-as-errors]
   24 | template <typename T> auto every_item_is_boolean(const T &container) -> bool {
      |                            ^
      |                       static

```

--- a/src/extension/alterschema/alterschema.cc
+++ b/src/extension/alterschema/alterschema.cc
@@ -7,21 +7,23 @@
 #include <cmath>
 #include <iterator>
 namespace sourcemeta::core {
-auto contains_any(const Vocabularies &container,
-                  const std::set<typename Vocabularies::key_type> &values)
-    -> bool {
+static auto
+contains_any(const Vocabularies &container,
+             const std::set<typename Vocabularies::key_type> &values) -> bool {
   return std::any_of(std::cbegin(container), std::cend(container),
                      [&values](const auto &element) {
                        return values.contains(element.first);
                      });
 }
 
-template <typename T> auto every_item_is_null(const T &container) -> bool {
+template <typename T>
+static auto every_item_is_null(const T &container) -> bool {
   return std::all_of(std::cbegin(container), std::cend(container),
                      [](const auto &element) { return element.is_null(); });
 }
 
-template <typename T> auto every_item_is_boolean(const T &container) -> bool {
+template <typename T>
+static auto every_item_is_boolean(const T &container) -> bool {
   return std::all_of(std::cbegin(container), std::cend(container),
                      [](const auto &element) { return element.is_boolean(); });
 }

